### DISCUSSION TITLE
[v3] Any way to detect whether a #[Reactive] property has been updated by the parent?

### DISCUSSION THREAD

**Discussion by jeffchown** [livewire/livewire#6636] [1. Help]
Hi all,

In v3, if I use the `#[Reactive]` attribute on a child component's property e.g. `public array $filters = [...]` and the parent component passes an updated `$filters` in the blade file e.g. `<livewire:child-component :filters="$filters" />`, is there a way to detect, in the child, whether it is being re-rendered due to the parent component updating its reactive property?

In the child component, I tried:

```
public function updated(string $name, mixed $value)
{
  ray('updated', $name, $value);
}
```

to the child component, but it is never called.

For example, my hope is to be able to call `$this->resetPage()` in the child component *only* if/when the parent updates any of the `$filters` array elements.

Thanks in advance for any ideas/help.

**Comment by jeffchown** [comment]
UPDATE: The more I think about it, there are multiple scenarios I might run in to (such as the one above) where I will simply need a way to detect if a child component is being rendered for the first time (e.g. handled by `mount()`) in the parent component or is being (re-)rendered due to a `#[Reactive]` property in the child component having been updated by/in the parent component.

I'm probably missing something, but Is there a way I can do this @calebporzio @joshhanley?

**Reply by whousehomestay** [reply]
any update on this?

**Comment by seedprod** [comment]
You can use a checksum in the render function like:
`        if($this->filter_checksum != md5(json_encode($this->filters)))
        {
            $this->resetPage();
        }
        $this->filter_checksum = md5(json_encode($this->filters));`

**Comment by DoneItInnovations** [comment]
Hello Folks.
I have a similar problem: 
In parent component I have a filter array. 
```
public array $filter = [
        'tags' => [],
        'position_categories' => [],
    ];
```
I pass this into the child components:
` <livewire:child :$filter />`
And inside the child component I use reactive on this:
```
#[Reactive]
public $filter;
```
I want check if $filter updated in the parent with updated hook instead of dispatch, but this is not runs because of some reason: 
```
function updatedFilter() 
{
      // I wrote this function in the child component, because I would like to trigger things here. In this example I want to pull down again the new filtered datas from API
}
```

PLEASE HELP :) @calebporzio @joshhanley

**Comment by whousehomestay** [comment]
For now, maybe this is the thing we can do: https://github.com/livewire/livewire/discussions/7989#discussioncomment-8551184
